spec/sqewer/active_job_spec.rb

```diff
@@ -92,4 +130,50 @@ class User < ActiveRecord::Base
     
     wait_for { user.reload.active? }.to eq(true)
   end
-end
+
+  it 'creates a file with option arguments and checks if it exists' do
+    wait_for { @worker.state }.to be_in_state(:running)
+
+    tmpdir = Dir.mktmpdir
```

Comment:
Shouldn't there be a cleanup step for all these temporary directories? If a block is not used, they [are not](http://ruby-doc.org/stdlib-2.0.0/libdoc/tmpdir/rdoc/Dir.html#method-c-mktmpdir) automatically cleaned up.

Reply:
Great suggestion and the Job for deleting files in unused. +1

Reply:
Temp folders usually are cleaned by a system at some point. On linux it is a boot time. OSX probably has the same behaviour.

Reply:
That's true, but I am on the principle that a test should not mutate anything/have side effects besides its context. It's safer and also avoids a lot of potential, hidden issues that may arise.